packages/trpc/server/routers/viewer/routing-forms/findTeamMembersMatchingAttributeLogic.handler.ts

```diff
@@ -0,0 +1,222 @@
+      message: "The route action is missing eventTypeId.",
+    });
+  }
+
+  const eventType = await EventTypeRepository.findByIdIncludeHosts({ id: route.action.eventTypeId });
+
+  if (!eventType) {
+    throw new TRPCError({
+      code: "NOT_FOUND",
+      message: "Event type not found",
+    });
+  }
+
+  const {
+    teamMembersMatchingAttributeLogic: matchingTeamMembersWithResult,
+    timeTaken: teamMembersMatchingAttributeLogicTimeTaken,
+    troubleshooter,
+    checkedFallback,
+    mainAttributeLogicBuildingWarnings: mainWarnings,
+    fallbackAttributeLogicBuildingWarnings: fallbackWarnings,
+  } = await findTeamMembersMatchingAttributeLogicOfRoute(
+    {
+      response,
+      route,
+      form: serializableForm,
+      teamId: form.teamId,
+      isPreview: !!isPreview,
+    },
+    {
+      enablePerf: _enablePerf,
+      // Reuse same flag for enabling troubleshooter. We would normall use them together
+      enableTroubleshooter: _enablePerf,
+      concurrency: _concurrency,
+    }
+  );
+
+  const urlSearchParams = getUrlSearchParamsToForwardForTestPreview({
+    formResponse: response,
+    fields: serializableForm.fields,
+    attributeRoutingConfig: route.attributeRoutingConfig ?? null,
+    teamMembersMatchingAttributeLogic: matchingTeamMembersWithResult
+      ? matchingTeamMembersWithResult.map((member) => member.userId)
+      : [],
+  });
+
+  const contactOwnerEmail = await getTeamMemberEmailForResponseOrContactUsingUrlQuery({
+    query: fromEntriesWithDuplicateKeys(urlSearchParams.entries()),
+    eventTypeId: eventType.id,
+    eventData: eventType,
+    chosenRoute: route,
+  });
+
+  if (!matchingTeamMembersWithResult) {
+    return {
+      contactOwnerEmail,
+      troubleshooter,
+      checkedFallback,
+      mainWarnings,
+      fallbackWarnings,
+      result: null,
+    };
+  }
+
+  const matchingTeamMembersIds = matchingTeamMembersWithResult.map((member) => member.userId);
+  const matchingTeamMembers = await UserRepository.findByIds({ ids: matchingTeamMembersIds });
+  const matchingHosts = eventType.hosts.filter((host) => matchingTeamMembersIds.includes(host.user.id));
+
+  if (matchingTeamMembers.length !== matchingHosts.length) {
+    throw new TRPCError({
+      code: "INTERNAL_SERVER_ERROR",
+      message: "Looks like not all matching team members are assigned to the event",
+    });
+  }
+
+  console.log("_enablePerf, _concurrency", _enablePerf, _concurrency);
+  if (_enablePerf) {
+    const serverTimingHeader = getServerTimingHeader(teamMembersMatchingAttributeLogicTimeTaken);
+    ctx.res?.setHeader("Server-Timing", serverTimingHeader);
+    console.log("Server-Timing", serverTimingHeader);
+  }
+  const { users: orderedLuckyUsers, perUserData } = matchingTeamMembers.length
+    ? await getOrderedListOfLuckyUsers(DistributionMethod.PRIORITIZE_AVAILABILITY, {
+        // Assuming all are available
+        availableUsers: [matchingTeamMembers[0], ...matchingTeamMembers.slice(1)],
```

Comment:
I think `matchingTeamMembers` is missing `weight` and `priority`

Reply:
Made `weight` and `priority` required fields with `null` values